src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -550,28 +550,60 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             End If
 
             Dim Here = 1
-            While CanGet(Here)
-                ch = Peek(Here)
-                If IsWhitespace(ch) Then
-                    Here += 1
+            Dim atNewLine As Boolean
+            ' Line continuation is valid at the end of the line, or at the end of the file, or followed by a trailing comment.
+            ' Eg.  LineContinuation ( EndOfLine | EndOfFile | LineContinuationComment)
+            If CanGet(Here) AndAlso (IsWhitespace(Peek(Here)) OrElse IsSingleQuote(Peek(Here))) Then
```

Comment:
>If [](start = 12, length = 2)

The second and third cases look very similar, so it feels like we can probably remove this outer condition and fold the third case into the second.
The code in the outer `Else` (`' We don't have a space or '`) would be removed. That case can be handled by the inner `Else` (`' If you get her you have a Line Continuation without comment, so process as V15.5 below`).

The logic would be: (1) skip whitespaces, (2) if single-quote do new processing, (3) if not single-quote, then capture the trivia and check if newline.

Reply:
@jcouv I will look at making the change tonight and test. This also might make it easier for @AnthonyDGreen to support Rem but I want to make sure I don't break anything. Thanks for the review.

Reply:
@jcouv  there are several corner cases covered by IDE test that need the logic as is. Several attempts to change the logic caused failures, one specific one on ubuntu I can't reproduce on Windows but it doesn't fail with original logic.

Reply:
@paul1956 can you expand on this?  It's unclear to me why Julien's suggestion would not work.  Namely, after getting the token, the logic for trailing trivia should be:

1. attempt to consume as much whitespace as possible.
2. attempt to consume a comment if possible.
3. consume a newline if possible

It's unclear what would be wrong with that.  Or why that would have a problem with something like `_EOF`.  If there is a problem, can you give an example of some code that doesn't work here, and why it doesn't work?

Reply:
_ whitespace is as scanner error (return false) and handled at a higher level where the trailing whitespace is removed but _ EOF is valid and is not expected to return false. If I consume all the whitespace then get to EOF the suggested change would report an error but the IDE expects no error. That explains multiple tests failing, but even when I fake it and return true one Ubuntu test still fails and I have no idea why.

Reply:
_ whitespace is as scanner error (return false) and handled at a higher level where the trailing whitespace is removed but _ EOF is valid and is not expected to return false. If I consume all the whitespace then get to EOF the suggested change would report an error but the IDE expects no error. That explains multiple tests failing, but even when I fake it and return true one Ubuntu test still fails and I have no idea why.

Reply:
I added that comment to the last commit after I undid the change.

Reply:
I added that comment to the last commit after I undid the change.

Reply:
@jcouv what is the next step to move forward.